hbase_storage/src/main/java/com/splicemachine/si/impl/server/PurgeConfigBuilder.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2012 - 2020 Splice Machine, Inc.
+ *
+ * This file is part of Splice Machine.
+ * Splice Machine is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3, or (at your option) any later version.
+ * Splice Machine is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.splicemachine.si.impl.server;
+
+public class PurgeConfigBuilder {
+    private PurgeConfig.PurgeLatestTombstone purgeLatestTombstone = null;
+    private Boolean respectActiveTransactions = null;
+    private Boolean purgeDeletes = null;
+    private Boolean purgeUpdates = null;
+
+    public PurgeConfigBuilder purgeDeletes(boolean b) {
+        assert purgeDeletes == null;
```

Comment:
why is it not allowed to tolerate situations like:
builder.purgeDeletes(true);
... // oops business logic shows that we can't do that
builder.purgeDeletes(false); // this will throw

Reply:
At the moment, we set all of that in the same place, so I'd rather be defensive about it and force users to only call that function once.